### PR TITLE
Few changes to Flash Grenade

### PR DESCRIPTION
**Pull request by woisalreadytaken**
Changes to Flash Grenade:
- No longer flashes people through walls. It's the whole point of this PR, as it kinda... makes sense. Doesn't really feel right for it to affect people who are taking cover. The rest of the PR is just added fluff.
- Added a short light effect when it explodes, for enemies taking cover nearby or teammates in general to see.
- Removed the default explosion sound from it. Looks like it was trying to play a different sound, but that didn't work half the time (and was always drowned out by the default explosion sound anyway). That's patched up now.
	- Also seems like frag grenades suffer from this same issue, but should be looked into in a different PR.

[Outdated preview](https://cdn.discordapp.com/attachments/332809334339665930/1002917306722177124/2022-07-30_09-34-40.mp4) - showcases flashes not going through walls

[Updated preview](https://cdn.discordapp.com/attachments/332809334339665930/1003216246814285835/2022-07-31_05-19-06.mp4) - showcases up-to-date light effects

(they're downloadable videos)

### CODE REVIEW

**Review by FortyTwoFortyTwo — APPROVED** [review]

File: addons/sourcemod/scripting/scp_sf/stocks.sp [hunk 1]
```diff
@@ -1694,4 +1694,64 @@ stock int TraceClientViewEntity(int client)
...
+	return entity;
+}
+
+public Action Timer_DestroyLight(Handle timer, int ref)
+{
+	int entity = EntRefToEntIndex(ref);
+	if (entity > MaxClients)
+	{
+		AcceptEntityInput(entity, "TurnOff");
+		RequestFrame(Frame_KillLight, ref);
```
Comment (Mikusch) [inline addons/sourcemod/scripting/scp_sf/stocks.sp:1746]: Is this necessary? I'd expect the entity to turn itself off and clean up after itself on removal.

File: addons/sourcemod/scripting/scp_sf/stocks.sp [hunk 2]
```diff
@@ -1694,4 +1694,64 @@ stock int TraceClientViewEntity(int client)
...
+	}
+	
+	return Plugin_Continue;
+}
+
+void Frame_KillLight(int ref)
+{
+	int entity = EntRefToEntIndex(ref);
+	if (entity > MaxClients)
+		AcceptEntityInput(entity, "Kill");
```
Comment (Mikusch) [inline addons/sourcemod/scripting/scp_sf/stocks.sp:1756]: Use `RemoveEntity`, same functionality but preferred

File: addons/sourcemod/scripting/scp_sf/stocks.sp (hunk 1 above)
Comment (Mikusch) [reply addons/sourcemod/scripting/scp_sf/stocks.sp:1746]: Sorry, wrong line. But I do question whether it's necessary to turn off the entity, and also to delay the removal.

Comment (woisalreadytaken) [reply addons/sourcemod/scripting/scp_sf/stocks.sp]: looks like it is not, I took the light stock from VSH and it was accompanied with this light-specific timer and delay, while also having a separate entity cleanup timer. Just thought they exist for a reason and never questioned it lol

**Review by Mikusch — APPROVED** [review]

**Pull request merged**